tpl/template.go

```diff
@@ -66,13 +68,16 @@ type TemplateDebugger interface {
 // TemplateAdapter implements the TemplateExecutor interface.
 type TemplateAdapter struct {
 	Template
+	Metrics metrics.Provider
 }
 
 // Execute executes the current template. The actual execution is performed
 // by the embedded text or html template, but we add an implementation here so
 // we can add a timer for some metrics.
 func (t *TemplateAdapter) Execute(w io.Writer, data interface{}) error {
-	// TODO(moorereason) metrics fmt.Println("Execute:", t.Name())
+	if t.Metrics != nil {
```

Comment:
I may read this wrong, but to me, it looks like `t.Metrics` is always set (i.e. a NoOpProvider if not enabled). So with that construct, you should get rid of this conditional ... but:

I would rather prefer that this was actual nil or zero or whatever (and drop the NoOp construct). It does not look expensive, but I would prefer not to create lots of time objects etc. in the normal run mode.

Reply:
Done.